internal/utils/write_file.go

```diff
@@ -0,0 +1,64 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+)
+
+/*
+WriteFile writes the given file contents to the given path.
+The file may optionally be marked as executable.
+
+In more detail:
+
+ 1. If a non-regular file exists at path, an error is returned.
+ 2. If a regular file exists at path, it is truncated and marked as executable if applicable.
+ 3. If no file exists at path, a new one is created with executable permissions if applicable
+
+In cases 2 and 3 above, the contents of scriptSource are then written to the file.
+*/
+func WriteFile(path string, contents []byte, executable bool) (err error) {
```

Comment:
Can we just use https://pkg.go.dev/io/ioutil#WriteFile instead?

Reply:
Or `os.WriteFile` with a small wrapper to fix perms.

Reply:
Originally I this function was only for executable script files, and I had a check that if the file was nonempty and nonexecutable, I'd return an error, as well as the one for non-regular files. I can remove the check for a non-regular file and unconditionally truncate if that's an acceptable simplification.